be/src/io/readable.h

```diff
@@ -0,0 +1,20 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#pragma once
+
+#include "common/statusor.h"
+
+namespace starrocks::io {
+
+// A Readable is a source of bytes. Bytes from a Readable are made available
+// to callers of the read method via a byte array.
+class Readable {
+public:
+    virtual ~Readable() = default;
+
+    // Read at most |count| bytes data into |data|.
+    // Returns the number of bytes read, 0 means EOF.
+    virtual StatusOr<int64_t> read(void* data, int64_t count) = 0;
```

Comment:
Wouldn't it be more convenient to provide an open and close interface?

Reply:
> Wouldn't it be more convenient to provide an open and close interface?

A `Readable` object is normally returned by an `open` operation of a specific implementation, for example, for a filesystem, you can `open` a file of a specific path, if the open succeeded, a Readable object will be returned, otherwise, an error is returned.
```
StatusOr<InputStream> open(std::string_view path);
```
So, there is no need to provide an `open` method in the abstract class `Readable`.  
Of course, an implementation of Readable can choose to provide an open method, but It's strange to require all
`Readable`s to have an ​open method.

The caller can release the Readable object after finishing reading the data, it's tedious to call a close a stream before releasing it.